crosscat/utils/inference_utils.py

```diff
@@ -124,30 +118,45 @@ def calculate_MI_bounded_discrete(X, Y, M_c, X_L, X_D):
         component_models_X[i] = cluster_models[X]
         component_models_Y[i] = cluster_models[Y]
 
-    MI = 0.0
+    def marginal_predictive_by_cluster(value, component_models):
+        return numpy.array([
+            component_models[j].calc_element_predictive_logp(value)
```

Comment:
Agreed.  Fixed in ef4fa1e1198b98b5ee47d88b1840319e1ef4bb9e .

On Thu, Oct 8, 2015 at 11:27 AM, gregory-marton <notifications@github.com>
wrote:

> In crosscat/utils/inference_utils.py
> <https://github.com/probcomp/crosscat/pull/60#discussion_r41527960>:
>
> > @@ -124,30 +118,45 @@ def calculate_MI_bounded_discrete(X, Y, M_c, X_L, X_D):
> >          component_models_X[i] = cluster_models[X]
> >          component_models_Y[i] = cluster_models[Y]
> >
> > -    MI = 0.0
> > +    def marginal_predictive_by_cluster(value, component_models):
> > +        return numpy.array([
> > +            component_models[j].calc_element_predictive_logp(value)
>
> I feel like either we should drop the _logp and _logps suffixes
> everywhere, or else use them everywhere. The mix is a tad confusing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/probcomp/crosscat/pull/60/files#r41527960>.
>